Spec/RealtimeClientChannel.swift

```diff
@@ -496,6 +496,82 @@ class RealtimeClientChannel: QuickSpec {
                         }
                     }
 
+                    // TO3g
+                    it("should immediately fail if not in the connected state") {
+                        let options = AblyTests.commonAppSetup()
+                        options.queueMessages = false
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        defer { client.dispose(); client.close() }
+                        let channel = client.channels.get("foo")
+                        waitUntil(timeout: testTimeout) { done in
+                            expect(client.connection.state).to(equal(.initialized))
+                            channel.publish(nil, data: "message") { error in
+                                expect(error).toNot(beNil())
+                                done()
```

Comment:
@paddybyers The publish should fail if the connection is in the INITIALIZED state, right? [`TO3g`](https://docs.ably.io/client-lib-development-guide/features/#TO3g) only mentions about connecting but it contradicts with [`RTL6c2`](https://docs.ably.io/client-lib-development-guide/features/#RTL6c2).

Reply:
cc @QuintinWillison